### PR TITLE
Support batch LowerCholeskyTransform

### PR DESCRIPTION
**Pull request by fehiepsi**
This pull request deals with batch LowerCholeskyTransform. I also add some tests and remove/change `variable` to `torch.tensor` in TestTransforms test.

### CODE REVIEW

File: test/test_distributions.py [hunk 1]
```diff
@@ -3020,23 +3020,23 @@ def setUp(self):
                 AbsTransform(cache_size=cache_size),
                 ExpTransform(cache_size=cache_size),
                 SigmoidTransform(cache_size=cache_size),
-                AffineTransform(variable(torch.Tensor(5).normal_()),
-                                variable(torch.Tensor(5).normal_()),
+                AffineTransform(torch.Tensor(5).normal_(),
+                                torch.Tensor(5).normal_(),
```
Comment (vishwakftw) [inline test/test_distributions.py]: These all have to be lowercase `tensor` I think.

**Review by apaszke — COMMENTED** [review]
LGTM apart from this minor thing. cc: @fritzo

File: torch/distributions/transforms.py [hunk 2]
```diff
@@ -468,12 +468,16 @@ class LowerCholeskyTransform(Transform):
...
         return x.tril(-1) + x.diag().exp().diag()
 
-    def _inverse(self, y):
-        if y.dim() != 2:
-            raise NotImplementedError
+    def _inverse_on_event(self, y):
         return y.tril(-1) + y.diag().log().diag()
+
+    def _call(self, x):
+        flat_x = x.view((-1,) + x.shape[-2:])
```
[the remark this answers is not in the document]
Comment (fehiepsi) [reply torch/distributions/transforms.py]: Yes, to make sure when users make a mistake of calling transform on noncontiguous tensor.

**Pull request merged**